### DISCUSSION TITLE
Plugin hooks

### DISCUSSION THREAD

**Discussion by vjpr** [oven-sh/bun#3069] [Q&A]
I would like a way to print how long each import takes to load.

The `Bun.plugin` api allows `onResolve` and `onLoad`. Is there something like `onLoadStart` / `onLoadEnd`. Or maybe a way to use `onLoad` and call the original loader...

Also, is there something like `Bun.Resolver` - to make use of Bun's resolver?

**Comment by vjpr** [comment]
https://bun.sh/docs/bundler/migration#plugin-api:

> The builder object provides some methods for hooking into parts of the bundling process. Bun implements onResolve and onLoad; it does not yet implement the esbuild hooks onStart, onEnd, and onDispose, and resolve utilities. initialOptions is partially implemented, being read-only and only having a subset of esbuild's options; use config (same thing but with Bun's BuildConfig format) instead.

I guess maybe we wait for these esbuild apis.

**Comment by ezekielchentnik** [comment]
any update on this? I'm in need of the additional hooks (specifically onEnd, module), I'm currently writing a svelte plugin and need these to extract css into a single file
